test/spec_session_abstract_persisted.rb

```diff
@@ -68,4 +68,30 @@ def session_exists?(req)
   it "#delete_session raises" do
     proc { @pers.send(:delete_session, nil, nil, nil) }.must_raise RuntimeError
   end
-end
+
+  it '#security_matches? returns true if secure cookie is off' do
+    @pers.send(:security_matches?, Rack::Request.new({}), {}).must_equal true
+  end
+
+  it '#security_matches? returns true if ssl is on' do
+    req = Rack::Request.new({})
+    req.set_header('HTTPS', 'on')
+    @pers.send(:security_matches?, req, { secure: true }).must_equal true
+  end
+
+  it '#security_matches? returns true if trusted_host option is set' do
+    req = Rack::Request.new({})
+    pers_with_persist = @class.new(nil, { trust_proxy: true })
+    pers_with_persist.send(:security_matches?, req, { secure: true }).must_equal true
+  end
+
+  it "#security_matches? returns true if host = 'localhost'" do
+    req = Rack::Request.new({})
+    req.set_header('HTTP_HOST', 'localhost')
+    @pers.send(:security_matches?, req, { secure: true }).must_equal true
+  end
+
+  it '#security_matches? returns true if secure cookie is off' do
+    @pers.send(:security_matches?, Rack::Request.new({}), { secure: true }).must_equal false
+  end
```

Comment:
Nit: please wrap these in a `describe "#security_matches?"` block. The other `it` blocks don't follow this pattern because there's only one of them per method.

Reply:
ok, done.